### PR TITLE
Whitelist envvars for jail

### PR DESCRIPTION
**Pull request by dramich**
Problem:
The jail clears the env and sets only ones required to run. This drops
envvars that might be needed for a proxy

Solution:
Add a whitelist of envvars to pull into the jail with a default list to
support proxies or user updated setting

Issue: https://github.com/rancher/rancher/issues/20709

### CODE REVIEW

File: pkg/jailer/jailer_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		},
+		{
+			input:    []string{"BAZ=FOOBAZ"},
+			expected: []string{"BAZ=FOOBAZ", "ENVVAR_FOO=ALPHA"},
+			envValue: "ALPHA",
+		},
+	}
+
+	for _, tc := range testCases {
+		os.Setenv("ENVVAR_FOO", tc.envValue)
```
Comment (daxmc99) [inline pkg/jailer/jailer_test.go]: Ridiculous nit but could the test cases be ordered:
{
envValue: "",
input:"",
expected:""
}

Comment (dramich) [reply pkg/jailer/jailer_test.go]: Done

File: pkg/jailer/jailer.go [hunk 2]
```diff
@@ -81,3 +83,20 @@ func GetUserCred() (*syscall.Credential, error) {
...
+
+func WhitelistEnvvars(envvars []string) []string {
+	wl := settings.WhitelistEnvironmentVars.Get()
+	envWhiteList := strings.Split(wl, ",")
+
+	if len(envWhiteList) == 0 {
+		return envvars
+	}
+
+	for _, wlVar := range envWhiteList {
```
Comment (alena1108) [inline pkg/jailer/jailer.go]: Do you think we should do `trimpSpace` on the each wlVar?

Comment (alena1108) [reply pkg/jailer/jailer.go]: Also do you know if Getenv is case sensitive? And can these variables be configured in a lower case on users' setups

Comment (dramich) [reply pkg/jailer/jailer.go]: Added trim space. Per discussion not going to do anything about case, user can utilize the setting if they need to change the case of the envvar being looked up.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**